legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/mixin/command/MixinMinecraft.java

```diff
@@ -0,0 +1,30 @@
+package io.github.minecraftcursedlegacy.mixin.command;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import io.github.minecraftcursedlegacy.api.command.DispatcherRegistry;
+import net.minecraft.client.Minecraft;
+import net.minecraft.level.Level;
+
+@Mixin(Minecraft.class)
+public class MixinMinecraft {
+	@Redirect(method = "tick()V", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/Minecraft;hasLevel()Z", ordinal = 0))
+	private boolean canOpenChatScreen(Minecraft minecraft) {
+		Level level = ((Minecraft) (Object) this).level;
```

Comment:
Why not shadow `level`?

Reply:
because mixin --migratemappings doesn't work and this can autoupdate

Reply:
unless you've added it to loom

Reply:
I suppose I'll have to now

Reply:
[There](https://github.com/Chocohead/Fabric-Loom/compare/7a685c...d5b560) you go then